frontend/public/components/utils/kebab.tsx

```diff
@@ -542,81 +529,75 @@ class KebabWithTranslation extends React.Component<
+          data-test-id="kebab-button"
+          disabled={isDisabled}
+          onClick={toggle}
+          onFocus={onHover}
+          onMouseEnter={onHover}
         >
-          <button
-            ref={this.dropdownElement}
-            type="button"
-            aria-expanded={this.state.active}
-            aria-haspopup="true"
-            aria-label={t('public~Actions')}
-            className="pf-c-dropdown__toggle pf-m-plain"
-            data-test-id="kebab-button"
-            disabled={isDisabled}
-            onClick={this.toggle}
-            onFocus={this.onHover}
-            onMouseEnter={this.onHover}
-          >
-            <EllipsisVIcon />
-          </button>
-          <Popper
-            open={!isDisabled && this.state.active}
-            placement="bottom-end"
-            closeOnEsc
-            closeOnOutsideClick
-            onRequestClose={this.handleRequestClose}
-            reference={this.getPopperReference}
+          <EllipsisVIcon />
+        </button>
+        <Popper
+          open={!isDisabled && active}
+          placement="bottom-end"
+          closeOnEsc
+          closeOnOutsideClick
+          onRequestClose={handleRequestClose}
+          reference={getPopperReference}
+        >
+          <FocusTrap
+            focusTrapOptions={{
+              clickOutsideDeactivates: true,
+              returnFocusOnDeactivate: false,
+              fallbackFocus: getDivReference, // fallback to popover content wrapper div if there are no tabbable elements
+            }}
           >
-            <FocusTrap
-              focusTrapOptions={{
-                clickOutsideDeactivates: true,
-                returnFocusOnDeactivate: false,
-                fallbackFocus: this.getDivReference, // fallback to popover content wrapper div if there are no tabbable elements
-              }}
-            >
-              <div ref={this.divElement} className="pf-c-dropdown pf-m-expanded" tabIndex={-1}>
-                <KebabMenuItems
-                  options={menuOptions}
-                  onClick={this.onClick}
-                  className="oc-kebab__popper-items"
-                  focusItem={menuOptions[0]}
-                />
-              </div>
-            </FocusTrap>
-          </Popper>
-        </div>
-      </Tooltip>
-    );
-  }
-}
+            <div ref={divElement} className="pf-c-dropdown pf-m-expanded" tabIndex={-1}>
+              <KebabMenuItems
+                options={menuOptions}
+                onClick={onClick}
+                className="oc-kebab__popper-items"
+                focusItem={menuOptions[0]}
+              />
+            </div>
+          </FocusTrap>
+        </Popper>
+      </div>
+    </Tooltip>
+  );
+};
+KebabWithTranslation.factory = kebabFactory;
+KebabWithTranslation.columnClass = 'dropdown-kebab-pf pf-c-table__action';
 
 function restoreStaticProperties(Kebab) {
   Kebab.factory = KebabWithTranslation.factory;
```

Comment:
Define the static props directly on the Kebab function component, remove the `restoreStaticProperties` function and the HOC pattern on L609. We can just export the FC directly since we are using the `useTranslation` hook now.
```suggestion
Kebab.factory = kebabFactory;
Kebab.columnClass = 'dropdown-kebab-pf pf-c-table__action';
Kebab.getExtensionsActionsForKind = getExtensionsKebabActionsForKind;
```